src/colp/components/usetype_version_comparison_report.py

```diff
@@ -0,0 +1,76 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import plotly.express as px
+from st_aggrid import AgGrid
+from src.constants import COLOR_SCHEME
+from abc import ABC
+import pdb
+
+
+class UsetypeVersionComparisonReport:
+    def __init__(self, data):
+        self.usetype_changes = data["usetype_changes"]
+
+    def __call__(self):
+        st.subheader("Usetype Version Comparison")
+
+        version = st.sidebar.selectbox(
+            "Select a Version for Comparison", ["2022/06/16"]
+        )
+
+        self.current_usetype_changes = self.filter_by_version(
+            self.usetype_changes, version
+        )
+
+        st.markdown(
+            f"""
+            This report compares the usetype distribution between the previous and selected versions.
+            Because there are a high number of usetypes, we have included a slider to be able to more clearly view a subsection.
+
+            **Current Version:** {version}
+
+            **Previous Version:** {self.previous_version(self.current_usetype_changes)}
+            """
+        )
+
+        usetype_range = st.slider(
+            "Select a range of values",
+            min_value=0,
+            max_value=self.current_usetype_changes.shape[0],
+            value=[0, 12],
+        )
+
+        df = self.sort_and_filter_df(self.current_usetype_changes, usetype_range)
+
+        self.display_graph(df, usetype_range)
+
+    def display_graph(self, df, range):
+        fig1 = px.bar(
+            data_frame=df,
+            y="usetype",
+            x=["difference", "num_records_current", "num_records_previous"],
+            barmode="group",
+            color_discrete_sequence=COLOR_SCHEME,
+            height=400 + (range[1] - range[0]) * 50,
+            width=1000,
+            text_auto=True,
+        )
+        fig1.update_xaxes(title="Count of Records")
+        fig1.update_yaxes(title="Use Type")
+        fig1.update_layout(legend_title_text="Version")
+
+        st.plotly_chart(fig1)
+
+    def filter_by_version(self, df, version):
+        return df.loc[df["v_current"] == version]
+
+    def previous_version(self, df):
+        return df.iloc[0]["v_previous"]
+
+    def sort_and_filter_df(self, df, range):
+        return (
+            df.sort_values(by="num_records_current", ascending=False)
```

Comment:
I needed a similar thing in my code to filter, limit records, and then filter again. Think it's more clear if it's three separate lines instead of chained dataframe method calls